File: examples/companion-packages/dotnet-fsharp-nuget-app/src/AgentXM.Examples.PawMatch.FSharp/MatchPreferences.fs
module AgentXM.Examples.PawMatch.FSharp.MatchPreferences

type T =
    { HasKids: bool
      QuietHome: bool
      Active: bool
      FirstTime: bool
      MultiplePets: bool
      SmallHome: bool }

let empty =
    { HasKids = false
      QuietHome = false
      Active = false
      FirstTime = false
      MultiplePets = false
      SmallHome = false }

let activeFlags prefs =
    seq {
        if prefs.HasKids then yield "has-kids"
        if prefs.QuietHome then yield "quiet-home"
        if prefs.Active then yield "active"
        if prefs.FirstTime then yield "first-time"
        if prefs.MultiplePets then yield "multiple-pets"
        if prefs.SmallHome then yield "small-home"
    }

let toFlagSet prefs = prefs |> activeFlags |> Set.ofSeq

let isEmpty prefs =
    not (
        prefs.HasKids
        || prefs.QuietHome
        || prefs.Active
        || prefs.FirstTime
        || prefs.MultiplePets
        || prefs.SmallHome
    )

File: examples/companion-packages/typescript-jsr-app/src/match_preferences.ts
export interface MatchPreferences {
  readonly hasKids: boolean;
  readonly quietHome: boolean;
  readonly active: boolean;
  readonly firstTime: boolean;
  readonly multiplePets: boolean;
  readonly smallHome: boolean;
}

export function activeFlagSet(prefs: MatchPreferences): Set<string> {
  const set = new Set<string>();
  if (prefs.hasKids) set.add("has-kids");
  if (prefs.quietHome) set.add("quiet-home");
  if (prefs.active) set.add("active");
  if (prefs.firstTime) set.add("first-time");
  if (prefs.multiplePets) set.add("multiple-pets");
  if (prefs.smallHome) set.add("small-home");
  return set;
}

export function isEmpty(prefs: MatchPreferences): boolean {
  return (
    !prefs.hasKids &&
    !prefs.quietHome &&
    !prefs.active &&
    !prefs.firstTime &&
    !prefs.multiplePets &&
    !prefs.smallHome
  );
}
